### PR TITLE
Adding "video upload" demo

### PR DESCRIPTION
**Pull request by WJ0417**
Please be free to leave any comment.

### CODE REVIEW

File: web_autoflip_demo/video_upload/video_upload.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+/*
```
Comment (parrajustin) [inline web_autoflip_demo/video_upload/video_upload.js:1]: If this is an auto-gen file from ts to js, add a line to the .gitignore to remove .js files

Comment (WJ0417) [reply web_autoflip_demo/video_upload/video_upload.js:1]: Sure, will do quickly!

Comment (WJ0417) [reply web_autoflip_demo/video_upload/video_upload.js]: Hi, Justin, I added the .gitignore and to remove the .js files, please check.

File: web_autoflip_demo/video_upload/video_upload.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. 
+*/
+
+//check if browser supports file api and filereader features
+if (window.File && window.FileReader && window.FileList && window.Blob) {
+  // this function is called when the input loads a video
+  function showVideo(): void {
+    var file = (<HTMLInputElement>event.target).files[0];
```
Comment (parrajustin) [inline web_autoflip_demo/video_upload/video_upload.ts:20]: Aligning with google style guide don't use var, use const with variables that aren't reassigned.

File: web_autoflip_demo/video_upload/video_upload.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    //grab the first video in the fileList
+    //Currently, only loading one file
+    console.log("video file has been chosen");
+    console.log(file);
+    renderVideo(file);
+  }
+
+  // this function is for displaying video content
+  function renderVideo(file: any): void {
+    var reader = new FileReader();
```
Comment (parrajustin) [inline web_autoflip_demo/video_upload/video_upload.ts:30]: Don't use var, use const

File: web_autoflip_demo/video_upload/video_upload.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    reader.onload = function (event) {
```
Comment (parrajustin) [inline web_autoflip_demo/video_upload/video_upload.ts:31]: I believe the google style guide calls for to put the type information, so replace "function (event) {" with "function (event: Event): void {". It's best to be explicit.

File: web_autoflip_demo/video_upload/video_upload.js (hunk 1 above)
Comment (parrajustin) [reply web_autoflip_demo/video_upload/video_upload.js]: Thanks for adding that!

File: web_autoflip_demo/video_upload/video_upload.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. 
+*/
+
+//check if browser supports file api and filereader features
+if (window.File && window.FileReader && window.FileList && window.Blob) {
```
[the remark this answers is not in the document]
Comment (WJ0417) [reply web_autoflip_demo/video_upload/video_upload.ts:17]: I strongly agree and will wait for Howard's opinion.

File: web_autoflip_demo/video_upload/video_upload.ts (hunk 2 above)
Comment (WJ0417) [reply web_autoflip_demo/video_upload/video_upload.ts:20]: fix

File: web_autoflip_demo/video_upload/video_upload.ts (hunk 3 above)
Comment (WJ0417) [reply web_autoflip_demo/video_upload/video_upload.ts:30]: fixed

File: web_autoflip_demo/video_upload/video_upload.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    console.log("video file has been chosen");
+    console.log(file);
+    renderVideo(file);
+  }
+
+  // this function is for displaying video content
+  function renderVideo(file: any): void {
+    var reader = new FileReader();
+    reader.onload = function (event) {
+      var the_url = event.target.result;
```
[the remark this answers is not in the document]
Comment (WJ0417) [reply web_autoflip_demo/video_upload/video_upload.ts:32]: fixed

File: web_autoflip_demo/video_upload/video_upload.js (hunk 1 above)
Comment (WJ0417) [reply web_autoflip_demo/video_upload/video_upload.js]: No problem!

File: web_autoflip_demo/video_upload/video_upload.ts (hunk 4 above)
Comment (WJ0417) [reply web_autoflip_demo/video_upload/video_upload.ts:31]: Thank you so much for teaching me the google style guide. For this line, I realized that I don't need a event parameter. So I removed event and get the URL from the reader. Please check the new commit.

File: web_autoflip_demo/video_upload/video_upload.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (hmulling) [reply web_autoflip_demo/video_upload/video_upload.ts:17]: I agree. We only need to support modern browsers.

[the remark this answers is not in the document]
Comment (WJ0417) [reply web_autoflip_demo/video_upload/video_upload.ts]: @parrajustin @hmulling Hi Howard and Justin, I remove the code about browser check in the latest commit. Please check and thank you so much for the instructions.

**Comment by parrajustin** [comment]
Looks good without any further comments I'll merge the pr.

**Comment by WJ0417** [comment]
Thank you so much for the quick reviewing and approval, I am so happy starting to have some code in the repo!

**Pull request merged**